Add BookEntity class

joejoh84 opened the pull request:
Closes #5

Vimbayinashe commented:
This looks great! I suggest that we add an ISBN field with a _unique_ constraint for BookEntity.

```
@Column(unique = true)
@NotBlank(message = "ISBN is a required field")
private String ISBN;
```

joejoh84 commented:
> This looks great! I suggest that we add an ISBN field with a _unique_ constraint for BookEntity.
> 
> ```
> @Column(unique = true)
> @NotBlank(message = "ISBN is a required field")
> private String ISBN;
> ```

Good idea including isbn. Updated entity to include isbn field and added to equals & hashcode

Vimbayinashe commented:
This looks great!
